tests/project/test_project_definition_v2.py

```diff
@@ -182,13 +180,48 @@ def test_project_definition_v2_schema(definition_input, expected_error):
                 raise err
 
 
+def test_identifiers():
+    definition_input = {
+        "definition_version": "2",
+        "entities": {
+            "A": {
+                "type": "application package",
+                "artifacts": [],
+                "manifest": "",
+            },
+            "B": {"type": "streamlit", "identifier": "foo_streamlit"},
+            "C": {
+                "type": "application",
+                "from": {"target": "A"},
+                "identifier": {"name": "foo_app", "schema": "schema_value"},
+            },
+            "D": {
+                "type": "application",
+                "from": {"target": "A"},
+                "identifier": {
+                    "name": "foo_app_2",
+                    "schema": "schema_value",
+                    "database": "db_value",
+                },
+            },
+        },
+    }
+    project = DefinitionV20(**definition_input)
```

Comment:
it might help ensure templating works fine if we use :
```    project = render_definition_template(definition_input, {}).project_definition``` instead